File: product/src/main/java/com/lexuancong/product/viewmodel/product/ProductDetailVm.java
package com.lexuancong.product.viewmodel.product;

import java.util.List;

public record ProductDetailVm(
        Long id,
        String name,
        String brandName,
        List<String> categories,
        List<AttributeGroupValueVm> attributeGroupValues,
        String shortDescription,
        String description,
        String specifications,
        Double price,
        boolean hasOptions,
        String avatarUrl,
        boolean isFeatured,
        List<String> images,
        boolean isOrderEnable
) {

}

File: frontend/models/product/productDetailVm.ts
import {AttributeGroupValueVm} from "@/models/product/attribute/AttributeGroupValueVm";

export  type  ProductDetailVm = {
    id: number,
    name: string,
    brandName: string,
    categories : string[],
    attributeGroupValues: AttributeGroupValueVm[],
    shortDescription: string,
    description: string,
    specifications: string,
    price : number,
    hasOptions : boolean,
    avatarUrl : string,
    isFeatured : boolean,
    isPublic: boolean,
    isOrderEnable  : boolean ,
    productImageUrls : string[]

}